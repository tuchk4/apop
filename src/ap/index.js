import crop from './actions/splice';
import filter from './actions/filter';
import join from './actions/join';
import map from './actions/map';
import remove from './actions/remove';
import sort from './actions/sort';
import swap from './actions/swap';

import { apply, recursiveApply } from './utils/apply';

const sourceActions = { crop, filter, join, map, remove, sort, swap };

const ap = Object.assign(apply, sourceActions);
const recursive = Object.assign(recursiveApply, sourceActions);

export default Object.assign(ap, { recursive });
