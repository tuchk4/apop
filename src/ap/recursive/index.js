import crop from '../actions/splice';
import filter from '../actions/filter';
import join from '../actions/join';
import map from '../actions/map';
import remove from '../actions/remove';
import sort from '../actions/sort';
import swap from '../actions/swap';

import { recursiveApply } from '../utils/apply';

const sourceActions = { crop, filter, join, map, remove, sort, swap };
const recursiveActions = {};

Object.keys(sourceActions).forEach(action => {
  recursiveActions[action] = (...args) =>
    recursiveApply(sourceActions[action](...args));
});

const recursive = Object.assign(
  (...actions) => recursiveApply(...actions),
  recursiveActions
);

export default recursive;
