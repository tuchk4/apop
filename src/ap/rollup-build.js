// import { rollup } from 'rollup';
//
// import resolve from 'rollup-plugin-node-resolve';
// import uglify from 'rollup-plugin-uglify';
// import commonjs from 'rollup-plugin-commonjs';
// import babel from 'rollup-plugin-babel';
// import { minify } from 'uglify-es';
// import babelrc from 'babelrc-rollup';
//
// const babelConfig = {
//   presets: [['es2015', { loose: true, modules: false }, 'stage-0']],
//   plugins: ['external-helpers', 'transform-object-rest-spread'],
// };
//
// var rollUpConfig = {
//   format: 'umd',
//   sourceMap: 'inline',
//   plugins: [
//     resolve({
//       jsnext: true,
//       main: true,
//       browser: true,
//     }),
//     commonjs(),
//     babel(
//       babelrc({
//         addExternalHelpersPlugin: false,
//         config: babelConfig,
//         exclude: 'node_modules/**',
//       })
//     ),
//   ],
// };
//
// async function make(config) {
//   let bundle = await rollup({
//     entry: `./src/${config.moduleName}.js`,
//     plugins: config.plugins,
//   });
//
//   await bundle.write({
//     format: config.format,
//     moduleName: config.moduleName,
//     dest: `./${config.moduleName}.js`,
//     sourceMap: config.sourceMap,
//   });
//   await bundle.write({
//     format: config.format,
//     moduleName: config.moduleName,
//     dest: `./${config.moduleName}.min.js`,
//     sourceMap: config.sourceMap,
//     plugins: [...rollUpConfig.plugins, uglify({}, minify)],
//   });
//   console.log(`make ./${config.moduleName}.js and ./${config.moduleName}.min.js`);
// }
//
// (async function() {
//
//   const builds = ['rmkArray','recursive/rmkArray'];
//
//   builds.forEach(async moduleName => await make({...rollUpConfig, moduleName}));
//
// })();
