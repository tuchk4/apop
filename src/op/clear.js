import action from './actions/clear';
import { apply } from './utils/apply';

export default config => apply(action(config));
