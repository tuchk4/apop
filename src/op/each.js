import action from './actions/each';
import { apply } from './utils/apply';

export default config => apply(action(config));
