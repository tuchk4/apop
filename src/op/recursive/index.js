import actions from '../actions';

import { recursiveApply } from '../utils/apply';

const recursive = Object.assign((...args) => recursiveApply(...args), actions);

export default recursive;
