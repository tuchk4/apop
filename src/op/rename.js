import action from './actions/rename';
import { apply } from './utils/apply';

export default config => apply(action(config));
