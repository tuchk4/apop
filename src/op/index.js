import clear from './actions/clear';
import toSnakeCase from './actions/toSnakeCase';
import toCamelCase from './actions/toCamelCase';
import update from './actions/update';
import rename from './actions/rename';
import each from './actions/each';

import { apply, recursiveApply } from './utils/apply';

const sourceActions = { clear, toSnakeCase, toCamelCase, update, rename, each };
const hackActions = {};

Object.keys(sourceActions).forEach(key => {
  hackActions[key] = (...args) =>
    Object.assign(sourceActions[key](...args), { shortcut: true, key });
});

const op = Object.assign((...args) => apply(...args), hackActions);
const recursive = Object.assign(
  (...args) => recursiveApply(...args),
  hackActions
);

export default Object.assign(op, { recursive });
