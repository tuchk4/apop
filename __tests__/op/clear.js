import ap from '../../src/build/ap';
import op from '../../src/build/op';

const before = Object.seal({
  id: 1,
  first_name: 'Stephen',
  last_name: 'Nelson',
  email: 'snelson0@yellowbook.com',
  gender: 'Male',
  ip_address: '74.191.108.58',
  city: 'Kapotnya',
  phrase: 'Universal 6th generation portal',
  date: new Date(Date.UTC(2001, 1, 1)),
  0: 0,
  1: 1,
  '-1': -1,
  emptyStr: '',
  null: null,
  undefined,
  nullArray: [null],
  undefinedArray: [undefined],
});

test('Clear object with config 1', () => {
  const formula = op.clear({
    nullValue: false,
    undefinedValue: false,
    emptyArray: false,
    emptyObject: false,
    emptyString: false,
  });

  const result = formula(before);
  expect(result).toMatchSnapshot();
});
test('Clear object with config 2', () => {
  const config = {
    nullValue: true,
    undefinedValue: true,
  };
  const formula = op.clear(config);

  const result = formula(before);
  expect(result).toMatchSnapshot();
});

test('Clear object with config 3', () => {
  const config = {
    emptyArray: true,
    emptyObject: true,
    emptyString: true,
  };
  const formula = op.clear(config);

  const result = formula(before);
  expect(result).toMatchSnapshot();
});

test('Clear object', () => {
  const formula = op.clear();
  const result = formula(before);
  expect(result).toMatchSnapshot();
});

test('Clear array of objects', () => {
  const formula = ap(ap.map(op.clear()));

  const result = formula([before]);
  expect(result).toMatchSnapshot();

  const formula2 = ap(op.clear());
  const result2 = formula2([before]);
  expect(result2).toMatchSnapshot();
});

test('Clear remove undefined and null values', () => {
  const result = op.clear()(before);
  expect(result).toMatchSnapshot();
});
