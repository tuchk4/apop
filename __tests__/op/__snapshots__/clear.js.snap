// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clear array of objects 1`] = `
Array [
  Object {
    "-1": -1,
    "0": 0,
    "1": 1,
    "city": "Kapotnya",
    "date": 2001-02-01T00:00:00.000Z,
    "email": "snelson0@yellowbook.com",
    "first_name": "Stephen",
    "gender": "Male",
    "id": 1,
    "ip_address": "74.191.108.58",
    "last_name": "Nelson",
    "phrase": "Universal 6th generation portal",
  },
]
`;

exports[`Clear array of objects 2`] = `
Array [
  Object {
    "-1": -1,
    "0": 0,
    "1": 1,
    "city": "Kapotnya",
    "date": 2001-02-01T00:00:00.000Z,
    "email": "snelson0@yellowbook.com",
    "first_name": "Stephen",
    "gender": "Male",
    "id": 1,
    "ip_address": "74.191.108.58",
    "last_name": "Nelson",
    "phrase": "Universal 6th generation portal",
  },
]
`;

exports[`Clear object 1`] = `
Object {
  "-1": -1,
  "0": 0,
  "1": 1,
  "city": "Kapotnya",
  "date": 2001-02-01T00:00:00.000Z,
  "email": "snelson0@yellowbook.com",
  "first_name": "Stephen",
  "gender": "Male",
  "id": 1,
  "ip_address": "74.191.108.58",
  "last_name": "Nelson",
  "phrase": "Universal 6th generation portal",
}
`;

exports[`Clear object with config 1 1`] = `
Object {
  "-1": -1,
  "0": 0,
  "1": 1,
  "city": "Kapotnya",
  "date": 2001-02-01T00:00:00.000Z,
  "email": "snelson0@yellowbook.com",
  "emptyStr": "",
  "first_name": "Stephen",
  "gender": "Male",
  "id": 1,
  "ip_address": "74.191.108.58",
  "last_name": "Nelson",
  "null": null,
  "nullArray": Array [
    null,
  ],
  "phrase": "Universal 6th generation portal",
  "undefined": undefined,
  "undefinedArray": Array [
    undefined,
  ],
}
`;

exports[`Clear object with config 2 1`] = `
Object {
  "-1": -1,
  "0": 0,
  "1": 1,
  "city": "Kapotnya",
  "date": 2001-02-01T00:00:00.000Z,
  "email": "snelson0@yellowbook.com",
  "emptyStr": "",
  "first_name": "Stephen",
  "gender": "Male",
  "id": 1,
  "ip_address": "74.191.108.58",
  "last_name": "Nelson",
  "nullArray": Array [
    null,
  ],
  "phrase": "Universal 6th generation portal",
  "undefinedArray": Array [
    undefined,
  ],
}
`;

exports[`Clear object with config 3 1`] = `
Object {
  "-1": -1,
  "0": 0,
  "1": 1,
  "city": "Kapotnya",
  "date": 2001-02-01T00:00:00.000Z,
  "email": "snelson0@yellowbook.com",
  "first_name": "Stephen",
  "gender": "Male",
  "id": 1,
  "ip_address": "74.191.108.58",
  "last_name": "Nelson",
  "null": null,
  "nullArray": Array [
    null,
  ],
  "phrase": "Universal 6th generation portal",
  "undefined": undefined,
  "undefinedArray": Array [
    undefined,
  ],
}
`;

exports[`Clear remove undefined and null values 1`] = `
Object {
  "-1": -1,
  "0": 0,
  "1": 1,
  "city": "Kapotnya",
  "date": 2001-02-01T00:00:00.000Z,
  "email": "snelson0@yellowbook.com",
  "first_name": "Stephen",
  "gender": "Male",
  "id": 1,
  "ip_address": "74.191.108.58",
  "last_name": "Nelson",
  "phrase": "Universal 6th generation portal",
}
`;
